api/scpca_portal/test/management/commands/test_create_portal_metadata.py

```diff
@@ -51,19 +52,41 @@ def test_create_portal_metadata(self):
         self.load_test_data()
         self.processor.create_portal_metadata(clean_up_output_data=False)
 
-        with ZipFile(common.OUTPUT_PORTAL_METADATA_ZIP_FILE_PATH) as zip:
+        zip_file_path = ComputedFile.get_local_file_path(
+            common.GENERATED_PORTAL_METADATA_DOWNLOAD_CONFIG
+        )
+        with ZipFile(zip_file_path) as zip:
             # Test the content of the generated zip file
-            # There is 1 file:
+            # There are 2 file:
             # ├── README.md
-            expected_file_count = 1
+            # |── metadata.tsv
+            expected_file_count = 2
             # The filenames should match the following constants specified for the computed file
-            expected_file = readme_file.OUTPUT_NAME
+            expected_files = {
+                readme_file.OUTPUT_NAME,
+                metadata_file.MetadataFilenames.METADATA_ONLY_FILE_NAME,
+            }
             files = set(zip.namelist())
             self.assertEqual(len(files), expected_file_count)
-            self.assertIn(expected_file, files)
+            self.assertEqual(files, expected_files)
+            for expected_file in expected_files:
+                self.assertIn(expected_file, files)
 
             # Test the content of README.md
             expected_text = (
                 "This download includes associated metadata for samples from all projects"
             )
             self.assertProjectReadmeContains(expected_text, zip)
+
+            # Test the content of metadata.tsv
+            tsv = (
+                zip.read(metadata_file.MetadataFilenames.METADATA_ONLY_FILE_NAME)
+                .decode("utf-8")
+                .splitlines()
+            )
```

Comment:
You should be able to open this with a combination of TextIOWrapper and csv.DictReader. Then you just need to ensure that the keys from the first object is a subset of `METADATA_COLUMN_SORT_ORDER`.

Reply:
See at 651d753c49c931eb71237155e64ad99226d6e9c8.